tools/myst/enc/enc.c

```diff
@@ -153,6 +153,17 @@ static size_t _get_num_tcs(void)
 
 int myst_setup_clock(struct clock_ctrl*);
 
+static void _sanitize_xsave_area_fields(uint64_t* rbx, uint64_t* rcx)
+{
+    assert(rbx && rcx);
+    /* replace XSAVE/XRSTOR save area size with fixed large value,
+    to protect against spoofing attacks from untrusted host.
+    If host returns smaller xsave area than required, this can cause a buffer
+    overflow at context switch time.
+    We believe value of 4096 should be sufficient for forseeable future */
+    *rbx = *rcx = 4096;
```

Comment:
Add sanity check to only override rbx, rcx, if the original value is less than 4096. This way, in case the code runs on some future CPU that reruns larger value, the implementation itself will not override the buffer size to be a value smaller than expected.

Reply:
Fixed.